File: Flutter/lib/features/chat/data/models/chat_model.dart
import '../../../message/data/models/message_model.dart';
import '../../domain/entities/chat.dart';
import 'member_model.dart';

class ChatModel extends Chat {
  const ChatModel({
    required super.id,
    required super.members,
    required super.name,
    required super.profile,
    required super.lastMessage,
    required super.group,
    required super.creator,
    required super.theme,
  });

  factory ChatModel.fromJson(Map<String, dynamic> json) => ChatModel(
        id: json['_id'] as String,
        members: (json['members'] as List<dynamic>)
            .map((e) => MemberModel.fromJson(e as Map<String, dynamic>))
            .toList(),
        name: json['name'] as String,
        profile: json['profile'] as String,
        lastMessage: MessageModel.fromJson(json['lastMessage'] as Map<String, dynamic>),
        group: json['group'] as bool,
        creator: json['creator'] as String,
        theme: json['theme'] as String,
      );

  Map<String, dynamic> toJson() => <String, dynamic>{
        '_id': id,
        'members': members,
        'name': name,
        'profile': profile,
        'lastMessage': lastMessage,
        'group': group,
        'creator': creator,
        'theme': theme,
      };
}

File: express/src/models/chat.model.ts
import mongoose from "mongoose";
import { IChat } from "../interfaces/IChat.js";

const ChatScheme = new mongoose.Schema(
  {
    members: [{ type: mongoose.Types.ObjectId, ref: "User" }],
    name: { type: String, default: null },
    profile: { type: String, default: null },
    lastMessage: {
      type: mongoose.Types.ObjectId,
      ref: "Message",
      default: null,
    },
    group: { type: Boolean, default: false },
    creator: {
      type: mongoose.Types.ObjectId,
      ref: "User",
      default: null,
    },
  },
  { timestamps: true }
);

const ChatModel = mongoose.model<IChat>("Chat", ChatScheme);

export default ChatModel;
